src/app/system/account-number-preferences/create-account-number-preference/create-account-number-preference.component.ts

```diff
@@ -0,0 +1,105 @@
+  /** Account Number Preferences Template Data */
+  accountNumberPreferencesTemplateData: any;
+  /** Prefix Type Data */
+  prefixTypeData: any[];
+
+  /**
+   * Retrieves the account number preferences template data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {SystemService} systemService Accounting Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private systemService: SystemService,
+              private route: ActivatedRoute,
+              private router: Router) {
+    this.route.data.subscribe((data: { accountnumberpreferencestemplate: any }) => {
+      this.accountNumberPreferencesTemplateData = data.accountnumberpreferencestemplate;
+    });
+  }
+
+  /**
+   * Creates the account number preference form.
+   * Subscribe on Form Controls to change Prefix Type data.
+   */
+  ngOnInit() {
+    this.createAccountNumberPreferenceForm();
+    this.changePrefixTypeValue();
+  }
+
+  /**
+   * Subscribes on Form Controls to change Prefix Type data.
+   */
+  changePrefixTypeValue() {
+    this.accountNumberPreferenceForm.get('addPrefix').valueChanges
+      .subscribe((addPrefix: boolean) => {
+        if (addPrefix) {
+          this.accountNumberPreferenceForm.get('prefixType').enable();
+          for (const accountType of this.accountNumberPreferencesTemplateData.accountTypeOptions) {
+            if (accountType.id === this.accountNumberPreferenceForm.get('accountType').value) {
+              this.prefixTypeData = this.accountNumberPreferencesTemplateData.prefixTypeOptions['accountType.' + accountType.value.toLowerCase()];
+              break;
+            }
+          }
+          this.accountNumberPreferenceForm.get('accountType').valueChanges
+            .subscribe((accountId: number) => {
+              for (const accountType of this.accountNumberPreferencesTemplateData.accountTypeOptions) {
+                if (accountType.id === accountId) {
+                  this.prefixTypeData = this.accountNumberPreferencesTemplateData.prefixTypeOptions['accountType.' + accountType.value.toLowerCase()];
+                  break;
+                }
+              }
+            });
+        } else {
+          this.accountNumberPreferenceForm.get('prefixType').disable();
+        }
+      });
+  }
+
+   /**
+    * Creates the create account number preference form.
+    */
+  createAccountNumberPreferenceForm() {
+    this.accountNumberPreferenceForm = this.formBuilder.group({
+      'accountType': ['', Validators.required],
+      'addPrefix': [false],
+      'prefixType': [{ value: '', disabled: true }]
+    });
+  }
+
+  /**
+   * Submits the account number preference form and creates a account number preference,
+   * if successful redirects to view account number preference.
+   */
+  submit() {
+    this.accountNumberPreferenceForm.get('addPrefix').disable();
```

Comment:
Why is this needed?

Reply:
It is a checkbox and its value should not be passed so that's why I am disabling it.

Reply:
Remove this checkbox as it is not required. Keep prefix as an optional field.

Reply:
Remove it from the edit component as well.